plugin.rb

```diff
@@ -47,4 +47,13 @@ module ::DiscourseTranslator
   add_to_serializer :post, :can_translate do
     scope.can_translate?(object)
   end
+
+  add_to_serializer :post, :cooked, respect_plugin_enabled: false do
+    return super() if cooked_hidden
+    DiscourseTranslator::TranslatorHelper.translated_value(super(), object, scope)
+  end
```

Comment:
Overriding a core's serializer attribute is going to be fragile to maintain long term. Perhaps we need an API extension in core?

Reply:
I've read through the post but it doesn't address my concern about us overriding a core's serializer attribute from a plugin which is a form of monkey patching. Just to clarify, my concern here is not about us changing the value of the `cooked` attribute but the approach in which we are doing so.

Reply:
I don't have strong opinions on whether 1 or 2 is the better approach. What I am saying is that if we are going with 1, we probably want to introduce a plugin modifier/API in Discourse core that allows us to modify the value of the `cooked` attribute in the serializer safely.

Reply:
Since this is experimental, I think we can add a note above the two `add_to_serializer` calls here, noting that we may/should refactor them before this can be production-ready.